javaslang/src/main/java/javaslang/concurrent/FutureImpl.java

```diff
@@ -193,7 +194,13 @@ void run(CheckedSupplier<? extends T> computation) {
                 // if the ExecutorService runs the computation
                 // - in a different thread, the lock ensures that the job is assigned before the computation completes
                 // - in the current thread, the job is already completed and the `job` variable remains null
-                final java.util.concurrent.Future<?> tmpJob = executorService.submit(() -> complete(Try.of(computation)));
+                final java.util.concurrent.Future<?> tmpJob = executorService.submit(() -> complete(Try.of(() -> {
+                    final long delta  = delay - System.currentTimeMillis() + start;
```

Comment:
For elapsed time measurements, should use System.nanoTime() and not the date-time APIs nor currentTimeMillis(). See https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#nanoTime-- for details -- mainly that nanoTime is guaranteed to be monotonic increasing (on most systems), whereas the date time APIs (which are based on currentTimeMillis) can move around during time changes and adjustments.